### DISCUSSION TITLE
Get data from checkbox

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [Textualize/textual#2851] [General]
Hi!

let's say I have this checkbox:
```
class CharacterSelection(App[None]):
    def compose(self) -> ComposeResult:
            with VerticalScroll():
                yield Center(Checkbox('0'))
                yield Center(Checkbox('1'))
                yield Center(Checkbox('2'))
                yield Center(Checkbox('3'))
                yield Center(Checkbox('4'))
                yield Center(Checkbox('5'))
     
??????????????????????????????????

```
I need the user to select only one option and get it, please tell me how to do it

**Comment by davep** [comment]
Looks to me like you want to be using a [RadioSet](https://textual.textualize.io/widgets/radioset/).

**Reply by IvanIsak2000** [reply]
Thanks for your reply! Yes, I really needed `RadioSet`.

I needed to get what the user chose, so I wrote my code like this:

```
    .....

    def on_radio_set_changed(self, event: RadioSet.Changed) -> None:
        self.exit(str(event.radio_set.pressed_index))
```

And then I equate this class to a variable and get the response index, perhaps this code will seem silly, is it possible to do it somehow wisely?

**Reply by davep** [reply]
Assuming that getting the value from the `RadioSet` is the fully extent of what you want the app for, and you want the app to close down when someone makes the selection, then yes, that's likely the best way to do this.

**Reply by IvanIsak2000** [reply]
Thank you very much!
